src/Polly.Shared/Timeout/TimeoutEngine.cs

```diff
@@ -47,8 +47,14 @@ internal static partial class TimeoutEngine
                         .Run(() =>
                             action(context, combinedToken)       // cancellation token here allows the user delegate to react to cancellation: possibly clear up; then throw an OperationCanceledException.
                             , combinedToken);           // cancellation token here only allows Task.Run() to not begin the passed delegate at all, if cancellation occurs prior to invoking the delegate.  
-
-                        actionTask.Wait(timeoutCancellationTokenSource.Token); // cancellation token here cancels the Wait() and causes it to throw, but does not cancel actionTask.  We use only timeoutCancellationTokenSource.Token here, not combinedToken.  If we allowed the user's cancellation token to cancel the Wait(), in this pessimistic scenario where the user delegate may not observe that cancellation, that would create a no-longer-observed task.  That task could in turn later fault before completing, risking an UnobservedTaskException.
+                        try
+                        {
+                            actionTask.Wait(timeoutCancellationTokenSource.Token); // cancellation token here cancels the Wait() and causes it to throw, but does not cancel actionTask.  We use only timeoutCancellationTokenSource.Token here, not combinedToken.  If we allowed the user's cancellation token to cancel the Wait(), in this pessimistic scenario where the user delegate may not observe that cancellation, that would create a no-longer-observed task.  That task could in turn later fault before completing, risking an UnobservedTaskException.
+                        }
+                        catch (AggregateException ex) when (ex.InnerExceptions.Count == 1)
+                        {
+                            throw ex.InnerException;
```

Comment:
Exactly: `ExceptionDispatchInfo.Capture().Throw()`.

The wrinkle is that .NET4.0 (one of our compile targets) does not support it.  We can consider solutions such as offered [here](https://stackoverflow.com/questions/20171877/is-there-an-analog-of-exceptiondispatchinfo-in-microsoft-bcl-async); whether we exclude .NET4.0 from this, ie do not fix #270 for .NET 4.0.